src/hepattn/callbacks/attn_mask_logger.py

```diff
@@ -0,0 +1,31 @@
+import matplotlib.pyplot as plt
+from lightning.pytorch.callbacks import Callback
+
+class AttnMaskLogger(Callback):
+    def on_validation_batch_end(self, trainer, pl_module, outputs, batch, batch_idx, dataloader_idx=0):
+        model = pl_module.model if hasattr(pl_module, "model") else pl_module
+        attn_mask_info = model.get_last_attention_mask() if hasattr(model, "get_last_attention_mask") else None
+        
+        if attn_mask_info is not None:
+            attn_mask, step, layer = attn_mask_info
+            plt.figure(constrained_layout=True, dpi=300)
+            plt.imshow(attn_mask.numpy(), aspect="auto")
```

Comment:
can you add a legend or colourbar to each image which indicates what `0` and `1` represent in terms of "used in attention", "masked", just so there is zero ambiguity